sdk/src/events.ts

```diff
@@ -0,0 +1,64 @@
+import {ContractTransaction} from 'ethers';
+import {Interface, LogDescription} from 'ethers/lib/utils';
+
+/**
+ * Finds a typed event in transaction given the event name
+ *
+ * @export
+ * @param {ContractReceipt} tx
+ * @param {string} eventName
+ * @return {*}  {(T)}
+ */
+export async function findEvent<T>(tx: ContractTransaction, eventName: string) {
+  const receipt = await tx.wait();
+
+  const event = (receipt.events || []).find(event => event.event === eventName);
+
+  if (!event) {
+    throw new Error(`No event found with name "${eventName}".`);
```

Comment:
I try to use a custom error class for errors so I dont need to compare the string to see what error is.

Reply:
Fixed in 44edb05c249276e4e8d6045a096b5c3cd12b6239